### PR TITLE
RGBDS syntax updates

### PR DESCRIPTION
**Pull request by vulcandth**
Please reference https://github.com/pret/pokecrystal/issues/686 and https://github.com/pret/pokecrystal/pull/905

If there is anything else that needs to be done as far as RGBDS goes, let me know.. and if I have time I can work on it.

### CODE REVIEW

File: engine/pinball_game/end_of_ball_bonus.asm [hunk 1]
```diff
@@ -543,15 +543,15 @@ Func_f8d5: ; 0xf8d5
 	ret
 
 AddBigBCD6: ; 0xf902
-x = 0
+DEF x = 0
 rept 6
```
Comment (dannye) [inline engine/pinball_game/end_of_ball_bonus.asm:547]: lol this should just be a for loop now

Comment (dannye) [inline engine/pinball_game/end_of_ball_bonus.asm]: lol this should just be a for loop now

File: engine/pinball_game/score.asm [hunk 2]
```diff
@@ -83,7 +83,7 @@ AddBigBCD6FromQueue: ; 0x8588
 	ld b, $1
 asm_8592:
 	push hl
-x = 0
+DEF x = 0
 rept 6
```
Comment (dannye) [inline engine/pinball_game/score.asm]: same here

File: home.asm [hunk 3]
```diff
@@ -3301,16 +3301,16 @@ BottomRightBonusStageCollisionMasks:
 
 ; These two squares data arrays must be aligned to $100 bytes and appear contiguously.
 SquaresLow:
-x = 0
+DEF x = 0
 rept 256
```
Comment (dannye) [inline home.asm]: same here

File: home.asm [hunk 4]
```diff
@@ -3301,16 +3301,16 @@ BottomRightBonusStageCollisionMasks:
...
 	db (x * x) % $100
-x = x + 1
+DEF x = x + 1
 endr
 
 SquaresHigh:
-x = 0
+DEF x = 0
 rept 256
```
Comment (dannye) [inline home.asm]: same here

File: home/random.asm [hunk 5]
```diff
@@ -131,8 +131,8 @@ EvensAndOdds:
 ; The first 128 bytes are the first 128 even numbers starting at 0.
 ; The next 128 bytes are the first 128 odd numbers starting at 1.
 ; The (a)th element is essentially what you'd get from rlca.
-x = 0
+DEF x = 0
 REPT 128
```
Comment (dannye) [inline home/random.asm]: same here

**Review by dannye — APPROVED** [review]
Thanks again! There are some old style `rept`s from before `for` was a thing that could be updated here if you wanted to, but no worries either way.
I'll also give @huderlem a chance to chime in first.

**Comment by huderlem** [comment]
> Thanks again! There are some old style `rept`s from before `for` was a thing that could be updated here if you wanted to, but no worries either way. I'll also give @huderlem a chance to chime in first.

I'm pretty unfamiliar with any new RGBDS features from the past few years.  I think Danny's suggestions are going to be sufficient.  Thanks!

File: engine/pinball_game/end_of_ball_bonus.asm (hunk 1 above)
Comment (vulcandth) [reply engine/pinball_game/end_of_ball_bonus.asm]: Easy enough; i'll take care of it when I get a chance tonight.

File: home/random.asm [hunk 6]
```diff
@@ -131,8 +131,8 @@ EvensAndOdds:
...
 ; The (a)th element is essentially what you'd get from rlca.
-DEF x = 0
-REPT 128
-	db x | ((x >> 7) & 1)
-DEF x = x + 2
+FOR X, 256
+	IF (X % 2 == 0)
+		db X | ((X >> 7) & 1)
+	ENDC
 ENDR
```
Comment (dannye) [inline home/random.asm]:
```suggestion
FOR X, 0, 256, 2
	db X | ((X >> 7) & 1)
ENDR
```

**Review by dannye — COMMENTED** [review]
just one little improvement

File: data/unused/unused_9800.asm [hunk 7]
```diff
@@ -1,19 +1,17 @@
...
-rept $100 / \1
-	db (x + y) & $ff
-y = y + \1
-endr
-x = x + 1
-endr
+MACRO macro_9800
+	FOR X, \1
+		DEF y = 0
+		rept $100 / \1
```
Comment (dannye) [inline data/unused/unused_9800.asm]: Oh sorry, one more `rept` that should be a `for`

Comment (vulcandth) [reply data/unused/unused_9800.asm]: Okay; pushed this.

**Review by dannye — APPROVED** [review]
Thanks! I'm good with this being merged whenever

**Pull request merged**